### PR TITLE
Dissolve on ImageMagick 6.9

### PR DESCRIPTION
**Pull request by jcrate**
ImageMagick 6.9 pulls the 'compose:args' artifact from the canvas image instead of the overlay image. The special_composite function now sets the 'compose:args' artifact in both images for 6.8 and 6.9 compatibility.

Added rspec test for Image#dissolve.

### CODE REVIEW

**Comment by jcrate** [comment]
Hopefully this can be included in a 3.2.1 version of the gem as well as the current 4.0.x release.

Also, I did not remove the Image#dissolve tests from test/Image2.rb, although most of those conditions are tested in the rspec version.

**Comment by mockdeep** [comment]
Hi @jcrate. Thanks so much for contributing a fix! It looks like there are some linter failures you'll need to handle before this can be merged. You might be able to run `rubocop -a` locally to handle them.

I wouldn't anticipate having a 3.2.1 release for this, though. Is there a reason you can't use 4.0?

**Comment by mockdeep** [comment]
https://travis-ci.org/rmagick/rmagick/jobs/558174323

**Comment by jcrate** [comment]
I updated the spec for rubocop, it should pass the linter. 

I use rmagick in a sizable project that's roughly 12 years old with little maintenance over the years, so I wasn't sure how much the removals in 4.0 would affect me. It turns out there wasn't much to do beyond the matte/alpha change, so I could potentially move to 4.0 now.

**Comment by mockdeep** [comment]
Thanks for updating! Yeah, you should be fine as long as you handled all of the deprecations in 3.2 and check over [the changelog](https://github.com/rmagick/rmagick/blob/master/CHANGELOG.md#rmagick-400) to make sure you aren't using any of the deprecated APIs. Depends on how deep your usage is, but I'd guess for 95% of users it should be a pretty painless transition.

I'm going to tag @Watson1978 and @dlemstra on this, since they've got a better sense of the C stuff.

File: ext/RMagick/rmimage.c [hunk 1]
```diff
@@ -1630,8 +1630,10 @@ special_composite(Image *image, Image *overlay, double image_pct, double overlay
     blend_geometry(geometry, sizeof(geometry), image_pct, overlay_pct);
     (void) CloneString(&overlay->geometry, geometry);
     (void) SetImageArtifact(overlay,"compose:args", geometry);
-
+
```
Comment (dlemstra) [inline ext/RMagick/rmimage.c]: Could you remove those extra spaces?

File: ext/RMagick/rmimage.c [hunk 2]
```diff
@@ -5293,7 +5295,7 @@ Image_dissolve(int argc, VALUE *argv, VALUE self)
 
     composite_image =  special_composite(image, overlay, src_percent, dst_percent
                                          , x_offset, y_offset, DissolveCompositeOp);
-
+
```
Comment (dlemstra) [inline ext/RMagick/rmimage.c]: And these?

**Pull request merged**